src/library.js

```diff
@@ -428,28 +428,47 @@ LibraryManager.library = {
     ___setErrNo(ERRNO_CODES.EINVAL);
     return -1;
   },
-  sbrk: function(bytes) {
+
+  // Implement a Linux-like 'memory area' for our 'process'.
+  // Changes the size of the memory area by |bytes|; returns the
+  // address of the previous top ('break') of the memory area
+  // We control the "dynamic" memory - DYNAMIC_BASE to DYNAMICTOP
+  sbrk__asm: true,
+  sbrk__sig: ['ii'],
+  sbrk__deps: ['__setErrNo'],
+  sbrk: function(increment) {
+    increment = increment|0;
+    var dynamicTop = 0;
+    increment = ((increment + 15) & -16)|0;
 #if USE_PTHREADS
-    if (ENVIRONMENT_IS_PTHREAD) return _emscripten_sync_run_in_main_thread_1({{{ cDefine('EM_PROXIED_SBRK') }}}, bytes);
+    dynamicTop = Atomics_add(HEAPU32, DYNAMICTOP_PTR>>2, increment);
+#else
+    dynamicTop = HEAPU32[DYNAMICTOP_PTR>>2]|0;
+    HEAPU32[DYNAMICTOP_PTR>>2] = dynamicTop + increment|0;
```

Comment:
That's a good point - I actually had to look up the precedence rules once just to verify. I'll add explicit parentheses here so there's no confusion.

Btw, raced with your comments to push a force update on this to fix some test breakages, sorry about that.